include/gsl/span

```diff
@@ -409,13 +409,21 @@ public:
 
     template <std::size_t N, class ArrayElementType = std::remove_const_t<element_type>>
     constexpr span(std::array<ArrayElementType, N>& arr) noexcept
-        : storage_(&arr[0], details::extent_type<N>())
+#if (defined(__cplusplus) && (__cplusplus >= 201703L))
```

Comment:
Apologies @annagrin, I lost your comment whilst addressing it by rebasing instead of merging (I'm used to VSTS).

@neilmacintosh feedback from 5/21
> Would std::data(arr) work as well? From memory, that is closer to the WD description of what happens.

@annagrin feedback from 6/15
> I just added tests for c++17. We can have conditional compilation dependent on __cplusplus value (with the idea that we eventually progress out of conditional compilation)